Utils/hdinsight-node-common/src/com/microsoft/azure/cosmosspark/serverexplore/cosmossparknode/CosmosSparkClusterNode.java

```diff
@@ -79,9 +81,12 @@ protected void updateNodeNameAfterLoading() {
     @Override
     protected void loadActions() {
         super.loadActions();
+
         String suffix = "/?adlaAccountName=" + adlAccount.getName();
-        addAction("Submit Job", new CosmosSparkSubmitAction(
-                this, cluster, CosmosSparkClusterOps.getInstance().getSubmitAction()));
+        NodeAction submitCosmosSparkJobAction = addAction(SUBMIT_COSMOS_SPARK_JOB_ACTION_NAME,
+                new CosmosSparkSubmitAction(this, cluster, CosmosSparkClusterOps.getInstance().getSubmitAction()));
+        submitCosmosSparkJobAction.setEnabled(isClusterStable());
```

Comment:
Could you change the presentation in `CosmosSparkSubmitAction`?

Reply:
`CosmosSparkSubmitAction` is of `NodeActionListener` type rather than `NodeAction` type, which means there is no `setEnabled()` method in `CosmosSparkSubmitAction`

---
In reply to: [240597908](https://github.com/Microsoft/azure-tools-for-java/pull/2487#discussion_r240597908) [](ancestors = 240597908)